lightning/src/ln/peer_handler.rs

```diff
@@ -378,6 +401,22 @@ impl Peer {
 			InitSyncTracker::NodesSyncing(pk) => pk < node_id,
 		}
 	}
+
+	/// Returns the number of onion messages we can fit in this peer's buffer.
+	fn onion_message_buffer_slots_available(&self) -> usize {
+		cmp::min(
```

Comment:
Please check out the latest implementation to make sure I understood. Planning to address the "only send OMs to channel peers" in follow-up

Reply:
Folks, have you thoughts about making those constants configurable with some `PerfConfig` to adjust to the node operator resources consumption policy ? Even if you have 1000 channels, 500 MiB of memory that can be too much. Rather to adopt an approach where we assume for X channels you have Y MiB per-outbound buffer, we could devolve the choice to the operator, like you have an global Z onion messages outbound buffer to allocate evenly or not (like first-come, first serve). 

Further, I think we should also care about ensuring the channel is well-confirmed or coming from a trusted peers, otherwise I could send you a thousand of `OpenChannel` messages and never broadcast the funding, if we allocate onion messages outbound bandwidth on the flight that could become a DoS Vector. I think same with dry-in up bandwidth for closed channels.

Could be all follow-ups.